src/monmove.cpp

```diff
@@ -359,22 +353,18 @@ void monster::move()
         moves = 0;
         return;
     }
-    if( friendly != 0 ) {
-        if( friendly > 0 ) {
-            friendly--;
-        }
-        friendly_move();
-        return;
+    if( friendly > 0 ) {
+        --friendly;
     }
 
     // Set attitude to attitude to our current target
     monster_attitude current_attitude = attitude( nullptr );
-    if( !plans.empty() ) {
-        if( plans.back() == g->u.pos3() ) {
+    if( !wander() ) {
```

Comment:
Not caused by this PR, just noting.
The section below is outdated and wrong: it doesn't test visibility and manually iterates over NPC instead of using `npc_at`. It should probably be modernized to use the more generic `Creature` attitude than just the one for `player` class.